### ISSUE TITLE
Database lock timer / unlock feature

### ISSUE THREAD

**Issue by bruntmyer29** [sqlitebrowser/sqlitebrowser#2466]
Wondering if there could be or already is a feature that could clear / Revert Changes after a certain amount of idle time? When sharing databases between users sometimes people will not realize that they have a pending change that is locking the database from others so it would be nice if there was a feature to automatically Revert Changes after a configured amount of time. 

Thanks,
Nick

![image](https://user-images.githubusercontent.com/52087523/97921951-51e4af80-1d21-11eb-91bb-d84116e205a4.png)

**Comment by MKleusberg** [comment]
Interesting idea! Just reverting the changes after a specified period wouldn't be too hard to implement. But I feel like there has to be a more subtle solution which doesn't just delete all your changes while you're having a lunch break :wink: However I can't come up with a working solution right now. Does anybody have an idea here?

**Comment by justinclift** [comment]
On the face of it, this sounds like we'd be opening up a can or worms and potential data-loss things. :frowning:

**Comment by chrisjlocke** [comment]
Indeed.  It would definitely have to be an option defaulted to 'false' 😄 
I've just had to answer a phone call ... I'd hate to return back to the PC to find DB4S had dropped all my changes as it got bored and stormed off in a huff! 😄

**Comment by bruntmyer29** [comment]
Yeah, by default it would need to be off. My thought was that it could be a feature that you'd need to turn on and set the amount of idle time you wanted. Maybe making some sort of additional popup instead of clearing all the data might be better? Just a thought as most issues I've run into with using this tool within a team is that the other user forgot they had pending updates they hadn't written yet. Which is turn keeps us all locked out of the database and some automated writes from being able to add data.

Thanks for looking at this,


Nick Bruntmyer

nbruntmy@hotmail.com<mailto:nbruntmy@hotmail.com>

Cell # 612-799-0784

________________________________
